server/src/main/java/com/objectcomputing/checkins/services/memberprofile/currentuser/CurrentUserController.java

```diff
@@ -0,0 +1,69 @@
+package com.objectcomputing.checkins.services.memberprofile.currentuser;
+
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.authentication.Authentication;
+import io.micronaut.security.rules.SecurityRule;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.net.URI;
+import java.util.*;
+
+@Controller("/services/member-profile/current")
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@Tag(name = "current user")
+public class CurrentUserController {
+
+    @Inject
+    CurrentUserServices currentUserServices;
+
+    /**
+     * Get user details from Google authentication
+     *
+     * @param authentication {@link Authentication} or null
+     * @return {@link HttpResponse<CurrentUserDTO>}
+     */
+    @Get
+    public HttpResponse<CurrentUserDTO> CurrentUser(@Nullable Authentication authentication) {
+
+        if (authentication == null) {
+            return HttpResponse
+                    .notFound();
+        }
+
+        String workEmail = authentication.getAttributes().get("email").toString();
```

Comment:
The default mapper will bring all these over? Works?

Reply:
Yes. Authentication.getAttributes() will return a Map<String, Object>. the following attributes are captured during SSO - 

{
at_hash=6LWYwacTRP2ndJ8PrBLpJQ, 
sub=109653345400507669053, 
email_verified=true, 
iss=https://accounts.google.com, 
given_name=Pramukh, 
locale=en, 
picture=https://lh3.googleusercontent.com/a-/AOh14GhwBEgfZ511QO2DLdLOpgSAkAkXnFBaSUxqlAMM=s96-c, 
aud=[832140020593-12q5hvojlj747n1vuupok5vvi995rec2.apps.googleusercontent.com], 
azp=832140020593-12q5hvojlj747n1vuupok5vvi995rec2.apps.googleusercontent.com, 
name=Pramukh Bagur, 
hd=objectcomputing.com, 
exp=Mon Aug 24 16:20:51 CDT 2020, 
family_name=Bagur, 
iat=Mon Aug 24 15:20:51 CDT 2020, 
email=bagurp@objectcomputing.com
}